editor/src/components/inspector/use-grid-advanced-properties.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react'
+import { getSimpleAttributeAtPath, MetadataUtils } from '../../core/model/element-metadata-utils'
+import { Substores, useEditorState } from '../editor/store/store-hook'
+import { isLeft, isRight, right } from '../../core/shared/either'
+import { isJSXElement } from '../../core/shared/element-template'
+import { styleP } from './inspector-common'
+
+/**
+ * These should match the props managed by the Advanced menu of the Grid inspector section.
+ */
+const advancedGridProps = [
+  styleP('justifyContent'),
+  styleP('alignContent'),
+  styleP('justifyItems'),
+  styleP('alignItems'),
+]
+
+export function useGridAdvancedPropertiesCount(): number {
+  const grid = useEditorState(
+    Substores.metadata,
+    (store) => {
+      if (store.editor.selectedViews.length !== 1) {
+        return null
+      }
+      const element = MetadataUtils.findElementByElementPath(
+        store.editor.jsxMetadata,
+        store.editor.selectedViews[0],
+      )
+      return MetadataUtils.isGridLayoutedContainer(element) ? element : null
+    },
+    'useGridAdvancedProperties grids',
+  )
+
+  return React.useMemo(() => {
+    if (grid == null || isLeft(grid.element) || !isJSXElement(grid.element.value)) {
+      return 0
+    }
+
+    const props = right(grid.element.value.props)
+
+    return advancedGridProps.reduce((acc, curr) => {
+      const attr = getSimpleAttributeAtPath(props, curr)
+      return isRight(attr) && attr.value != null ? acc + 1 : acc
```

Comment:
It would nice to create a `count` function in array-utils for this (for the note, a filter + length could work too)

Reply:
Could use an optic and `lengthOf`.

Reply:
factored out to `array-utils.ts` in [`f525303` (#6468)](https://github.com/concrete-utopia/utopia/pull/6468/commits/f5253036906cf00be01fa74543059c3ffc0230d5), but then moved it `element-metadata-utils.ts` in [`a6d3d2d` (#6468)](https://github.com/concrete-utopia/utopia/pull/6468/commits/a6d3d2dc6884b6a32029e588432b0cb53c9356ab) because of a failing import loop